src/components/FTP/FTPconnect.tsx

```diff
@@ -1,85 +1,141 @@
+import { useCallback, useState, useEffect } from "react";
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
-import { UserGuide } from "../UserGuide/UserGuide";
 import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { RenderComponent } from "../UserGuide/UserGuide";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
 
-const title = "FTP Tool";
-const description_userguide =
-    "A quick guide on how to use the FTP on Deakin-Detonator-Toolkit:\n" +
-    "- Type in FTP server's IP address on the box\n" +
-    "- If it prompts you with a login, type your login details\n" +
-    "- If it does not, type in user and then type in your login details\n" +
-    "- Type lcd to change the current directory to the default directory. \n Example: /home/kali\n" +
-    "- Enjoy using ftp with DDT tools :)";
-
-// Values user will input into form
+/**
+ * Represents the form values for the FTPConnect component.
+ */
 interface FormValuesType {
-    IPaddress: string;
+    ipAddress: string;
 }
 
-export function FTPconnect() {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
+/**
+ * The FTPconnect component.
+ * @returns The FTPconnect component.
+ */
+function FTPconnect() {
+    // Component State Variables
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+
+    // Component Constants.
+    const title = "FTP"; // Title of the component.
+    const description = "FTP tool"; // Description of the component.
+    const steps =
+        "Step 1: Type in FTP server's IP address.\n" +
+        "Step 2: If prompted with a login, type login details, if not type in 'user' and then type in login details.\n" +
```

Comment:
Add "the" and split into two sentences for clarity. 
```suggestion
        "Step 2: If prompted with a login, type the login details. If not prompted, type in 'user' and then type in login details.\n" +
```